hikari/impl/entity_factory.py

```diff
@@ -1913,6 +1916,15 @@ def serialize_command_option(self, option: commands.CommandOption) -> data_bindi
         if option.options is not None:
             payload["options"] = [self.serialize_command_option(suboption) for suboption in option.options]
 
+        if option.min_value is not None:
+            payload["min_value"] = (
+                math.ceil(option.min_value) if option.type is commands.OptionType.INTEGER else option.min_value
+            )
+        if option.max_value is not None:
+            payload["max_value"] = (
+                math.floor(option.max_value) if option.type is commands.OptionType.INTEGER else option.max_value
```

Comment:
Maybe its better to document the behavior when using `INTEGER` or `NUMBER` and just assume people read documentation. Discord will probably error if this happens, so it wont be as ominous as just it being casted upwards or downwards

Reply:
I'd say it's always better to gracefully handle floats since for the user the outcome would always be the same as if floats were allowed. Might be a good idea to recommend using integers in the docs though.